apps/arm/main.cpp

```diff
@@ -180,7 +187,25 @@ static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-    return MBED_SUCCESS; // TODO
+    // ASSUME PAYLOAD IS 32 BITS LONG
+    int32_t payload(0);
```

Comment:
wouldn't the payload be a 32 bit floating point number? PID gains are typically floating point values, not integer ones.

 floating point math is fine since we have an FPU

Reply:
i couldn't do this bc the payload is really a set of three numbers. the first is an integer which is an id. the second is a bit indicating whether we are changing velocity or position pid. then there's a floating point number. the 3rd case was awkward bc payload is an int and it's likely a float. So i think i will just divide by 1000.0 to make it a float with 4 decimal precision. Like if someone wants the param to be 1.25, the can packet will have 12500, which is an int.

Reply:
I mean the data segment of a can frame can be up to 64 bits long

32bit floating point gain value
3bits joint id
1bit vel or pos mode
2 bits to indicate whether changing kp, ki or kd
27 bits where u can store other stuff

Reply:
that 65 lol but yeah i'll do that